.github/workflows/commit-format.yml

```diff
@@ -41,4 +52,32 @@ jobs:
             fi
           done
           exit $retval
-     
+
+  check_commit_signoff:
+    runs-on: ubuntu-latest
+    name: check-commit-signoff
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Check for `Signed-off-by` line
+        run: |
+          if echo ${{ secrets.FSF_CONTRIBUTORS }} | grep ${{ github.event.pull_request.user.login }}; then
+            echo "author has an FSF copyright assignment"
+          fi
+
+          retval=0
+          for commit in $(git rev-list origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.sha }});
+          do
+            echo -n "Checking commit sign-off for $commit: "
+            if [[ $(git log $commit -1 --format="%B" | tail -n 2) = Signed-off-by:* ]]; then
```

Comment:
Also need to consider `Co-authored-by` tags (any case)?

Reply:
Should they really be considered? I don't mind writing the shell to do so, but it does seem maybe a bit overkill considering we have very little Co-authored-by commits. If you think it's necessary I'll do it :)